mimic/test/test_core.py

```diff
@@ -55,3 +58,43 @@ def test_from_plugin_includes_all_plugins(self):
             len(plugin_apis),
             len(list(core.entries_for_tenant('any_tenant', {},
                                              'http://mimic'))))
+
+    def test_load_domain_plugin_includes_all_domain_plugins(self):
+        """
+        Using the :func:`MimicRoot.fromPlugin` creator for a
+        :class:`MimicCore`, domains mocks implementing `class`:`IAPIDomainMock`
+        are included.
+        """
+        self.root = FilePath(self.mktemp())
+        self.root.createDirectory()
+        self.package = self.root.child('fakeplugins')
+        self.package.createDirectory()
+        init = b"""from fakeplugins._domain import domain_plugin
+__all__ = [domain_plugin]
+"""
+        self.package.child('__init__.py').setContent(init)
+        plugin = b"""from mimic.test.dummy import ExampleDomainAPI
+domain_plugin = ExampleDomainAPI()
+__all__ = [domain_plugin]
+"""
+        self.package.child('_domain.py').setContent(plugin)
+        sys.path.insert(1, self.root.path)
+
+        import fakeplugins
+        import mimic.plugins
+
+        self.original = mimic.plugins
```

Comment:
I really dislike how I'm doing this. I'm working on trying to use `sys.modules["mimic.plugins"] = fakeplugins`, yet I've not gotten this work yet.